python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -90,7 +91,10 @@ def __init__(
                 tasks, or ``"actors"`` to use an autoscaling actor pool.
             ray_remote_args: Args to provide to ray.remote.
         """
-        name = f"{name}({_get_udf_name(fn)})"
+        if inspect.ismethod(fn) and isinstance(fn.__self__, Preprocessor):
+            name = fn.__self__.__class__.__name__
```

Comment:
copies over the logic here (will remove this code later when we remove the actual stages code) https://github.com/ray-project/ray/blob/master/python/ray/data/dataset.py#L631-L636

Reply:
right now, we expect the op name to be just the class name `<PreprocessorClassName>`, and if we omit this special case, this would become `MapBatches(<PreprocessorClassName>)`. to maintain consistency with the existing code, i propose we keep this custom logic for now. i can move all of the logic into `_get_udf_name`